profile-specific-challenges/frontend/components/idioms.jsx

```diff
@@ -52,12 +58,13 @@ export function UseEffect({ fetchURL, label }) {
 }
 
 export function UseEffectDerivedCalculation() {
-  const [remainder, setReminder] = useState()
   const [clickedTimes, setClickedTimes] = useState()
 
-  useEffect(() => {
-    setReminder(clickedTimes % 5)
-  }, [clickedTimes])
+  // memoizing this might be an overkill.
+  // the useEffect will trigger the reconciliation algorithm twice, one after setting the clickedItems state
+  // and one after setting the remainder state.
+  // we saved some commits with this derived state
+  const remainder = useMemo(() => clickedTimes % 5, [clickedTimes]);
```

Comment:
Should not memo this, its an unnecessary memory hodg. The key is to not cause re-renders for cheap calculations.

Reply:
You are right; I overlooked the logic